### PR TITLE
Improve tracing architecture documentation

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Improve the diagram and description 
- Move the `search for traces` to a separate document (may need a follow up)

**Related issue(s)**
See also #4790

### CODE REVIEW

File: docs/tracing/02-01-tracing.md [hunk 1]
```diff
@@ -2,45 +2,33 @@
...
-![Request traces](./assets/request-traces.svg)
 
-## Store traces
+## Collect traces
 
-Traces are stored in Jaeger in the following way:
+The process of collecting traces by Jaeger looks as follows:
+ 
+1. The application receives trace data.
+2. Istio Proxy configured for the application makes sure to propagate the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests further to the `zipkin` service, responsible for looking up and collecting traces from various distributed systems.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
2. If the application has istio-injection enabled, the Istio Proxy for the application makes sure to propagate the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests further to the Jaeger deployment accessing it via the `zipkin` service, exposing a Jaeger port compatible to the zipkin protocol.
```

File: docs/tracing/02-01-tracing.md [hunk 2]
```diff
@@ -2,45 +2,33 @@
...
+3. Zipkin receives the data and passes it further through [Istio Proxy](https://github.com/istio/proxy) which provides further authorization mechanisms to ensure proper communication.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion

```

File: docs/tracing/02-01-tracing.md [hunk 3]
```diff
@@ -2,45 +2,33 @@
...
-## Store traces
+## Collect traces
 
-Traces are stored in Jaeger in the following way:
+The process of collecting traces by Jaeger looks as follows:
+ 
+1. The application receives trace data.
+2. Istio Proxy configured for the application makes sure to propagate the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests further to the `zipkin` service, responsible for looking up and collecting traces from various distributed systems.
+3. Zipkin receives the data and passes it further through [Istio Proxy](https://github.com/istio/proxy) which provides further authorization mechanisms to ensure proper communication. 
+4. Jaeger processes the data. Specifically, the  `jaeger-agent` service receives the spans, batches them and forwards to collectors.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
4. Jaeger processes the data. Specifically, the  `agent`-component receives the spans, batches them and forwards to the `collectors?-component. 
```

File: docs/tracing/02-01-tracing.md [hunk 4]
```diff
@@ -2,45 +2,33 @@
...
+## Collect traces
 
-Traces are stored in Jaeger in the following way:
+The process of collecting traces by Jaeger looks as follows:
+ 
+1. The application receives trace data.
+2. Istio Proxy configured for the application makes sure to propagate the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests further to the `zipkin` service, responsible for looking up and collecting traces from various distributed systems.
+3. Zipkin receives the data and passes it further through [Istio Proxy](https://github.com/istio/proxy) which provides further authorization mechanisms to ensure proper communication. 
+4. Jaeger processes the data. Specifically, the  `jaeger-agent` service receives the spans, batches them and forwards to collectors. 
+5. The data is then saved in the database.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
5. The data is then saved in a database (BadgerDB) persisted on a persistent volume.
```

File: docs/tracing/02-01-tracing.md [hunk 5]
```diff
@@ -2,45 +2,33 @@
...
-2. Envoy passes the trace details to the Zipkin Kubernetes service. This service acts as a facade receiving the trace and span details.
-3. The Zipkin service forwards the tracing information to Jaeger Deployment, allowing it to process them.
+## Query traces
 
-![Store traces](./assets/store-traces.svg)
+The process of querying traces from Jaeger looks as follows:
 
-## Search traces by tags
+1. A Kyma user accesses the Jaeger UI to [look for specific traces](/components/tracing#details-search-for-traces).
+2. Jaeger UI passes the request to the `jaeger-query` service. The requests goes through the [Istio Ingress Gateway](https://kyma-project.io/docs/components/application-connector/#architecture-application-connector-components-istio-ingress-gateway) which receives and processes the incoming connections between the service and the external source.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
2. Jaeger UI passes the request to the `jaeger-query` service. The requests goes through the [Istio Ingress Gateway](https://kyma-project.io/docs/components/application-connector/#architecture-application-connector-components-istio-ingress-gateway) which forwards the incoming connections to the service.
```

File: docs/tracing/02-01-tracing.md [hunk 6]
```diff
@@ -2,45 +2,33 @@
...
+3. The `jaeger-query` service passes the request to the [Keycloak Gatekeeper](https://github.com/keycloak/keycloak-gatekeeper) for verification. The Gatekeeper calls [Dex](https://github.com/dexidp/dex) to authenticate the user and the request, and grants further access if the authentication is successful.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
3. The `jaeger-query` service passes the request to the [Keycloak Gatekeeper](https://github.com/keycloak/keycloak-gatekeeper) for request authorization. The Gatekeeper calls [Dex](https://github.com/dexidp/dex) to authenticate the user and the request, and grants further access if the authentication is successful. 
```

File: docs/tracing/02-01-tracing.md [hunk 7]
```diff
@@ -2,45 +2,33 @@
...
+## Query traces
 
-![Store traces](./assets/store-traces.svg)
+The process of querying traces from Jaeger looks as follows:
 
-## Search traces by tags
+1. A Kyma user accesses the Jaeger UI to [look for specific traces](/components/tracing#details-search-for-traces).
+2. Jaeger UI passes the request to the `jaeger-query` service. The requests goes through the [Istio Ingress Gateway](https://kyma-project.io/docs/components/application-connector/#architecture-application-connector-components-istio-ingress-gateway) which receives and processes the incoming connections between the service and the external source.
+3. The `jaeger-query` service passes the request to the [Keycloak Gatekeeper](https://github.com/keycloak/keycloak-gatekeeper) for verification. The Gatekeeper calls [Dex](https://github.com/dexidp/dex) to authenticate the user and the request, and grants further access if the authentication is successful. 
+4. The request goes through [Istio Proxy](https://github.com/istio/proxy) which provides further authorization mechanisms to ensure proper communication.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion

```

File: docs/tracing/02-01-tracing.md [hunk 8]
```diff
@@ -2,45 +2,32 @@
...
-![Request traces](./assets/request-traces.svg)
 
-## Store traces
+## Collect traces
 
-Traces are stored in Jaeger in the following way:
+The process of collecting traces by Jaeger looks as follows:
+ 
+1. The application receives a request, either from an internal or external source.
+2. If the application has Istio injection enabled, [Istio proxy](https://github.com/istio/proxy) makes sure to propagate the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests further to the Jaeger Deployment. Istio proxy accesses the Deployment using the [Zipkin](https://zipkin.io/) service which exposes a Jaeger port compatible with the Zipkin protocol.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/tracing/02-01-tracing.md]: Does Istio proxy propagate the correct headers itself? (then we could get rid of `make sure`)

File: docs/tracing/02-01-tracing.md [hunk 9]
```diff
@@ -2,45 +2,32 @@
...
+In Kyma, the tracing component based on Jaeger provides the necessary functionality to collect and query traces. Both operations may occur at the same time, meaning that you can inspect specific traces using the Jeager UI, while at the same time Jaeger takes care of proper trace collection and storage. See the diagram for details: 
 
-## Request traces
-
-The process of requesting traces from Jaeger looks as follows:
-
-1. A Kyma user accesses Jaeger UI.
-2. The user uses the UI to request the trace details for a given service by selecting the service from the **Services** drop-down menu and confirming the choice by selecting the **Find Traces** button. Jaeger passes the request to jaeger-query, which is the UI facade.
-3. The jaeger-query forwards the details to Jaeger Deployment. The kcproxy verifies each request and if the authentication is successful, Jaeger sends the requested information back.
+![Tracing architecture](./assets/tracing-architecture.svg)
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/tracing/02-01-tracing.md]: I noticed that Jaeger is capitalized and sometimes not - not sure if that is how it should be?
image

File: docs/tracing/02-01-tracing.md [hunk 10]
```diff
@@ -2,45 +2,31 @@
...
-![Request traces](./assets/request-traces.svg)
 
-## Store traces
+## Collect traces
 
-Traces are stored in Jaeger in the following way:
+The process of collecting traces by Jaeger looks as follows:
+ 
+1. The application receives a request, either from an internal or external source.
+2. If the application has Istio injection enabled, [Istio proxy](https://github.com/istio/proxy) propagates the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests to the Jaeger Deployment. Istio proxy accesses the Deployment using the [Zipkin](https://zipkin.io/) service which exposes a Jaeger port compatible with the Zipkin protocol.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
2. If the application has Istio injection enabled, [Istio proxy](https://github.com/istio/proxy) propagates the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests to the Jaeger Deployment. Istio proxy calls Jaeger using the [Zipkin](https://zipkin.io/) service which exposes a port compatible with the Zipkin protocol.  
```

File: docs/tracing/02-01-tracing.md [hunk 11]
```diff
@@ -2,45 +2,31 @@
...
-## Store traces
+## Collect traces
 
-Traces are stored in Jaeger in the following way:
+The process of collecting traces by Jaeger looks as follows:
+ 
+1. The application receives a request, either from an internal or external source.
+2. If the application has Istio injection enabled, [Istio proxy](https://github.com/istio/proxy) propagates the correct [HTTP headers](/components/tracing#details-propagate-http-headers) of the requests to the Jaeger Deployment. Istio proxy accesses the Deployment using the [Zipkin](https://zipkin.io/) service which exposes a Jaeger port compatible with the Zipkin protocol.  
+3. Jaeger processes the data. Specifically, the  Jaeger Agent component receives the spans, batches them, and forwards to the Jaeger Collector service. 
+4. The BadgerDB database stores the data and persists it using the PersistentVolume resource.
```
Comment (a-thaler) [inline docs/tracing/02-01-tracing.md]:
```suggestion
4. The BadgerDB database stores the data and persists it using a PersistentVolume resource.
```

**Pull request merged**